### PR TITLE
Follow Security Guide to update release.yml

### PR DESCRIPTION
**Pull request by ActoryOu**
Follow Security Guide to update release.yml

Description
-----------

Follow [Security Guide](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#understanding-the-risk-of-script-injections) to update release.yml.

Test Steps
-----------

Execute release flow in local branch, see result by https://github.com/ActoryOu/FreeRTOS-Kernel/actions/runs/11626810540/job/32379084241.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [NA] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/auto-release.yml [hunk 1]
```diff
@@ -44,37 +44,49 @@ jobs:
...
-          git checkout -b ${{ github.event.inputs.version_number }} ${{ github.event.inputs.commit_id }}
+          git checkout -b "$VERSION_NUMBER" "$COMMIT_ID"
           echo "COMMIT_SHA_1=$(git rev-parse HEAD)" >> $GITHUB_ENV
 
       - name: Update source files with version info
+        env:
+          VERSION_NUMBER: ${{ github.event.inputs.version_number }}
+          MAIN_BR_VERSION_NUMBER: ${{ github.event.inputs.main_br_version }}
+          COMMIT_SHA_1: ${{ env.COMMIT_SHA_1 }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (kstribrnAmzn) [inline .github/workflows/auto-release.yml:67]: Is the `GITHUB_TOKEN` env variable necessary? Seems like it's unused.

File: .github/workflows/auto-release.yml [hunk 2]
```diff
@@ -83,24 +95,32 @@ jobs:
...
+          git push -u origin "$VERSION_NUMBER"
           echo "COMMIT_SHA_2=$(git rev-parse HEAD)" >> $GITHUB_ENV
 
       - name: Release
+        env:
+          VERSION_NUMBER: ${{ github.event.inputs.version_number }}
+          MAIN_BR_VERSION_NUMBER: ${{ github.event.inputs.main_br_version }}
+          COMMIT_SHA_2: ${{ env.COMMIT_SHA_2 }}
+          REPO_OWNER: ${{ github.repository_owner }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (kstribrnAmzn) [inline .github/workflows/auto-release.yml:113]: Same as above - the `GITHUB_TOKEN` variable looks unused.

File: .github/workflows/auto-release.yml (hunk 1 above)
Comment (ActoryOu) [reply .github/workflows/auto-release.yml:67]: Refer to [result](https://github.com/ActoryOu/FreeRTOS-Kernel/actions/runs/11633404737/job/32398481898), this is used in py script.

> Traceback (most recent call last):
>   File "/home/runner/work/FreeRTOS-Kernel/FreeRTOS-Kernel/./tools/.github/scripts/update_src_version.py", line 173, in <module>
>     main()
>   File "/home/runner/work/FreeRTOS-Kernel/FreeRTOS-Kernel/./tools/.github/scripts/update_src_version.py", line 152, in main
>     assert 'GITHUB_TOKEN' in os.environ, 'Set env{GITHUB_TOKEN} to an authorized git PAT'
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> AssertionError: Set env{GITHUB_TOKEN} to an authorized git PAT

File: .github/workflows/auto-release.yml (hunk 2 above)
Comment (ActoryOu) [reply .github/workflows/auto-release.yml:113]: Refer to [result](https://github.com/ActoryOu/FreeRTOS-Kernel/actions/runs/11633424178/job/32398542219), this is used in py script.

> Traceback (most recent call last):
>   File "/home/runner/work/FreeRTOS-Kernel/FreeRTOS-Kernel/./tools/.github/scripts/release.py", line 625, in <module>
>     main()
>   File "/home/runner/work/FreeRTOS-Kernel/FreeRTOS-Kernel/./tools/.github/scripts/release.py", line 576, in main
>     assert 'GITHUB_TOKEN' in os.environ, 'Set env{GITHUB_TOKEN} to an authorized git PAT'
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> AssertionError: Set env{GITHUB_TOKEN} to an authorized git PAT
> Error: Process completed with exit code 1.

**Review by kstribrnAmzn — DISMISSED** [review]

**Review by kar-rahul-aws — APPROVED** [review]

**Review by thomas-roos — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**